MaxText/configs/a3/llama_2_7b/16vm.sh

```diff
@@ -1,35 +1,47 @@
+# --device-type ${DEVICE_TYPE} --num-nodes 16 \
 # --command "bash MaxText/configs/a3/llama_2_7b/16vm.sh"
 
 # Stop execution if any command exits with error
 set -e
 
 export OUTPUT_PATH="gs://maxtext-experiments-multipod"
-export RUN_NAME="llama-2-16vm-$(date +%Y-%m-%d-%H-%M)"
 
 # Set environment variables
 for ARGUMENT in "$@"; do
     IFS='=' read -r KEY VALUE <<< "$ARGUMENT"
     export "$KEY"="$VALUE"
 done
 
-export XLA_FLAGS="--xla_dump_to=$OUTPUT_PATH/$RUN_NAME/HLO_dumps/
---xla_gpu_enable_latency_hiding_scheduler=true
---xla_gpu_enable_triton_gemm=false --xla_gpu_graph_level=0
---xla_gpu_enable_highest_priority_async_stream=true
---xla_gpu_all_reduce_combine_threshold_bytes=1073741824 --xla_gpu_all_gather_combine_threshold_bytes=134217728
---xla_gpu_reduce_scatter_combine_threshold_bytes=134217728 --xla_gpu_enable_pipelined_all_gather=true
---xla_gpu_enable_pipelined_reduce_scatter=true --xla_gpu_enable_pipelined_all_reduce=true
---xla_gpu_enable_while_loop_double_buffering=true --xla_gpu_enable_triton_softmax_fusion=false
---xla_gpu_enable_all_gather_combine_by_dim=false --xla_gpu_enable_reduce_scatter_combine_by_dim=false
---xla_disable_hlo_passes=rematerialization"
+# The setup accommodates two cases:
+# 1) Passing the 'RUN_NAME' variable at runtime
+# 2) Propagating the 'M_RUN_NAME' variable within an Airflow sweeping workflow
+if [ -n "$RUN_NAME" ];
+then
+    export M_RUN_NAME=$RUN_NAME
+fi
+
+export XLA_FLAGS="--xla_dump_to=$OUTPUT_PATH/$M_RUN_NAME/HLO_dumps/ --xla_dump_hlo_pass_re=.*
+ --xla_gpu_enable_latency_hiding_scheduler=true
+ --xla_gpu_enable_triton_gemm=false --xla_gpu_graph_level=0
+ --xla_gpu_enable_highest_priority_async_stream=true
+ --xla_gpu_all_reduce_combine_threshold_bytes=1073741824 --xla_gpu_all_gather_combine_threshold_bytes=134217728
+ --xla_gpu_reduce_scatter_combine_threshold_bytes=134217728 --xla_gpu_enable_pipelined_all_gather=true
+ --xla_gpu_enable_pipelined_reduce_scatter=true --xla_gpu_enable_pipelined_all_reduce=true
+ --xla_gpu_enable_while_loop_double_buffering=true --xla_gpu_enable_triton_softmax_fusion=false
+ --xla_gpu_enable_all_gather_combine_by_dim=false --xla_gpu_enable_reduce_scatter_combine_by_dim=false
+ --xla_disable_hlo_passes=rematerialization"
 
-# 16 nodes
-python MaxText/train.py MaxText/configs/base.yml run_name=$RUN_NAME hardware=gpu \
+# 16 node, DCN_DP=16, ICI_FSDP=8
+python MaxText/train.py MaxText/configs/base.yml hardware=gpu \
     steps=30 dcn_data_parallelism=16 ici_fsdp_parallelism=8 per_device_batch_size=4 max_target_length=4096 model_name=llama2-7b \
     enable_checkpointing=false attention=cudnn_flash_te remat_policy=minimal_flash use_iota_embed=true scan_layers=false \
-    dataset_type=synthetic async_checkpointing=false base_output_directory=gs://runner-maxtext-logs profiler=xplane
+    dataset_type=synthetic async_checkpointing=false base_output_directory=$OUTPUT_PATH logits_dot_in_fp32=false profiler=xplane
```

Comment:
Have you tested the logits_dot_in_fp32 change, e.g. via a golden logits test?

Reply:
Actually, we have been using `logits_dot_in_fp32=false` for both A3 and A3+ benchmarking, but this flag was missing when we added these scripts. Would you point me which golden logits test shall we run?

Reply:
hmm I do think a golden logits test makes sense to include, it will be giving coverage to both GPUs in general and to these specific choices